aider.el

```diff
@@ -381,6 +402,78 @@ The command will be formatted as \"/ask \" followed by the text from the selecte
   :lighter " Aider"
   :keymap aider-minor-mode-map)
 
+(defvar aider-prog-mode-map (make-sparse-keymap)
+  "Keymap for Aider commands in programming modes.")
+
+;;;###autoload
+(define-minor-mode aider-prog-mode
+  "Minor mode for Aider programming mode integration."
+  :lighter " Aider"
+  :keymap aider-prog-mode-map)
+
+;;;###autoload
+(defun aider-setup-doom-bindings ()
+  "Set up Doom-specific keybindings for Aider in programming modes."
+  (when (and (featurep 'evil) (fboundp 'map!))
+    (map! :map aider-prog-mode-map
+          :leader
+          (:prefix ("l" . "aider")
+                   (:prefix ("l" . "buffer")
+                    :desc "aider-run-aider (buffer)" "o" #'aider-run-aider
+                    :desc "aider-clear" "c" #'aider-clear
+                    :desc "aider-reset" "r" #'aider-reset
+                    :desc "aider-exit" "x" #'aider-exit)
+
+                   (:prefix ("a" . "add")
+                    :desc "aider-add-current-file" "c" #'aider-add-current-file
+                    :desc "aider-add-files-in-current-window" "w" #'aider-add-files-in-current-window
+                    :desc "aider-batch-add-dired-marked-files" "b" #'aider-batch-add-dired-marked-files
+                    :desc "aider-repo-find-name-dired" "g" #'aider-repo-find-name-dired
+                    :desc "aider-git-repo-root-dired" "d" #'aider-git-repo-root-dired)
+
+                   (:prefix ("r" . "read")
+                    :desc "aider-current-file-read-only" "f" #'aider-current-file-read-only
+                    :desc "aider-send-line-under-cursor" "l" #'aider-send-line-under-cursor
+                    :desc "aider-send-paragraph" "p" #'aider-send-paragraph)
+
+                   (:prefix ("c" . "change")
+                    :desc "aider-code-change" "c" #'aider-code-change
+                    :desc "aider-region-refactor" "r" #'aider-region-refactor
+                    :desc "aider-undo-last-change" "u" #'aider-undo-last-change)
+
+                   (:prefix ("d" . "discuss")
+                    :desc "aider-ask-question" "a" #'aider-ask-question
+                    :desc "aider-architect-discussion" "d" #'aider-architect-discussion
+                    :desc "aider-region-explain" "r" #'aider-region-explain
+                    :desc "aider-debug-exception" "e" #'aider-debug-exception)
+
+                   (:prefix ("o" . "other")
+                    :desc "aider-general-command" "c" #'aider-general-command
+                    :desc "aider-help" "h" #'aider-help
+                    :desc "aider-magit-show-last-commit" "g" #'aider-magit-show-last-commit)))))
+
+
+;;;###autoload
+(defun aider-maybe-enable-prog-mode ()
+  "Enable aider-prog-mode if we're in a programming mode and Doom bindings are enabled."
+  (when (and aider-enable-doom-bindings
+             (derived-mode-p 'prog-mode))
```

Comment:
My personal feeling is aider can help modify document file, for example, the README.org. Not sure if we want to limit it to prog-mode?

Reply:
Yes, I had a similar feeling. I actually wanted to update this to instead check if they are in a git repo. That's seems to be the requirement for aider, so it could make sense here, too. What do you think?